spec/immutable-data-structures.html

```diff
@@ -568,6 +568,40 @@ <h1>Tuple.prototype.unshifted ( ..._args_ )</h1>
           1. Return a new Tuple value whose [[Sequence]] is _list_.
         </emu-alg>
       </emu-clause>
+      <emu-clause id="sec-array.prototype.tolocalestring">
+        <h1>Tuple.prototype.toLocaleString ( [ _reserved1_ [ , _reserved2_ ] ] )</h1>
+        <p>An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the `Tuple.prototype.toLocaleString` method as specified in the ECMA-402 specification. If an ECMAScript implementation does not include the ECMA-402 API the following specification of the `toLocaleString` method is used.</p>
+        <p>The meanings of the optional parameters to this method are defined in the ECMA-402 specification; implementations that do not include ECMA-402 support must not use those parameter positions for anything else.</p>
+        <p>When the *toLocaleString* method is called, The following steps are taken:</p>
+        <emu-alg>
+          1. Let _T_ be ? thisTupleValue(*this* value).
+          1. Let _separator_ be the String value for the list-separator String appropriate for the host environment's current locale (this is derived in an implementation-defined way).
+          1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-tuple-element-tolocalestring-functions" title></emu-xref>.
+          1. Let _mapper_ be ! CreateBuiltinFunction(_steps_, &laquo; &raquo;).
+          1. Return ? TupleToString(_T_, _separator_, _mapper_).
+        </emu-alg>
+
+        <emu-clause id="sec-tuple-element-tolocalestring-functions">

```

Comment:
very much winging it here, would love some more input. Because records don't have a prototype, we won't have a `toLocaleString` to call on a Record object inside a Tuple. I'm special casing this here to use a mapping function that handles Record-type elements specifically, so that Tuples with Records won't fail on `toLocaleString`.

Is this the right way to do this?

Reply:
I don't see why you should handle Tuples with Records in them specially; I think it's fine if this fails in that case. We're just trying to be parallel to Arrays, not trying to make everything magical. I think the most important case here is really lists of numbers.

Reply:
Makes sense to me. The simplest solution here is to instead just copy over %Array.prototype.toLocaleString% in this case, which I've done here.